bentoml/deployment/aws_lambda/utils.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import shutil
+import subprocess
+import logging
+import re
+
+import boto3
+from botocore.exceptions import ClientError
+
+from bentoml.exceptions import BentoMLException, BentoMLDeploymentException
+from bentoml.utils.whichcraft import which
+
+logger = logging.getLogger(__name__)
+
+
+AWS_LAMBDA_APP_PY_TEMPLATE_HEADER = """\
+import os
+
+# Set BENTOML_HOME to /tmp directory due to AWS lambda disk access restrictions
+os.environ['BENTOML_HOME'] = '/tmp/bentoml/'
+from bentoml.deployment.aws_lambda.utils import download_artifacts_for_lambda_function
+from bentoml.bundler import load_bento_service_class
+
+tmp_artifacts_dir = '/tmp/bentoml'
+s3_bucket = '{s3_bucket}'
```

Comment:
Recommend read s3_bucket, artifacts_prefix, and bento_name from env var and move this to a separate python file that we can test.

About the api_name below, one workaround is to set the api_name is use seattr to current module:
```
import sys

api_name = os.environ.get("BENTOML_API_NAME")
thismodule = sys.modules[__name__]

def api_func(event, context):
    api = bento_service.get_service_api(api_name)
    return api.handle_aws_lambda_event(event)

setattr(thismodule, 'BENTOML_API_NAME', api_func)
```